docs/proposals/api_design.md

```diff
@@ -0,0 +1,62 @@
+---
+title: "HTTP API Design"
+linkTitle: "HTTP API Design"
+weight: 1
+slug: http-api-design
+---
+
+# HTTP API Design
+
+## Overview
+
+The purpose of this design document is to propose a set of standards that should be the basis of the Cortex HTTP API. This document will outline the current state of the Cortex http api and describe limitations that result from the current approach. It will also outline a set of paradigms on how http routes should be created within Cortex.
+
+## Current Design
+
+As things currently stand, the majority of HTTP API calls exist under the `/api/prom` path prefix. This prefix is configurable. However, since this prefix is shared between all the modules which leads to conflicts if the Alertmanager is attempted to be run as as part of the single binary.
+
+## Proposed Design
```

Comment:
Does your proposed design keeps the `-http.prefix` support? I think there's some value in being able to configure a prefix which applies to all routes, but the default should be empty string.

Reply:
I agree. However, I would like to keep that flag as is and add a deprecated warning to it's usage and still route to the endpoints as they exists under that prefix. I think a new set of flags can be added to prefix the new API.